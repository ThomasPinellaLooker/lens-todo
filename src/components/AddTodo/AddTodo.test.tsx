import { mount } from 'enzyme';
import { theme, ThemeProvider } from 'looker-lens';
import * as React from 'react'
import { create } from 'react-test-renderer';
import "../../../test/setupJest.js"
import { AddTodo } from './AddTodo';

test('Default AddTodo', () => {
  const addTodo = () => undefined

  const component = create(
    <ThemeProvider theme={theme}>
      <AddTodo addTodo={addTodo} />
    </ThemeProvider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Add button is disabled initially', () => {
  const addTodo = () => undefined

  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <AddTodo addTodo={addTodo} />
    </ThemeProvider>
  )

  const addButton = wrapper.find('button')
  expect(addButton.props().disabled).toBe(true)
})
