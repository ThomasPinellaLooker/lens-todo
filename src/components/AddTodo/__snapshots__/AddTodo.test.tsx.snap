// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default AddTodo 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-EHOje empeAK sc-bdVaJa jkGTWO"
  >
    <div
      className="todo-input-container sc-bdVaJa eWGIUr"
    >
      <div
        className="sc-gzVnrw euKcLj"
      >
        <label
          className="sc-htoDjs bUgUqE"
          htmlFor={undefined}
        />
        <div
          className="sc-EHOje empeAK sc-bdVaJa bZoazI"
        >
          <div
            className="sc-bZQynM dqpfBg sc-bdVaJa eWGIUr"
          >
            <input
              className="sc-gZMcBi eLaHKN"
              onChange={[Function]}
              placeholder="Add a todo..."
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <button
      className="sc-bwzfXH hzkIez sc-bdVaJa cIzdnm"
      disabled={true}
    >
      Add it
    </button>
  </div>
</form>
`;
